components/dashboard/src/theme-context.tsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import React, { createContext, useEffect, useState } from 'react';
+
+export const ThemeContext = createContext<{
+    isDark?: boolean,
+    setIsDark: React.Dispatch<boolean>,
+}>({
+    setIsDark: () => null,
+});
+
+export const ThemeContextProvider: React.FC = ({ children }) => {
+    const [ isDark, setIsDark ] = useState<boolean>(document.documentElement.classList.contains('dark'));
+    const actuallySetIsDark = (dark: boolean) => {
+      document.documentElement.classList.toggle('dark', dark);
+      setIsDark(dark);
+    }
+
+    useEffect(() => {
+        const observer = new MutationObserver(() => {
+            if (document.documentElement.classList.contains('dark') !== isDark) {
```

Comment:
out of curiosity, why do we need an observer for `documentElement`?

Reply:
That's in order to detect when the `classList` changes (receives `dark` class, or `dark` class is removed). This is the main control for light vs dark, and we want to listen to this in order to re-render components that have a programatic dependency on theme (e.g. to swap out an image).

Notes:
- Most of the time, it's better to not listen to `isDark`, but instead use appropriate CSS classes (e.g. `dark:bg-gray-900`)
- Also, this is different from the "theme" preference (which has 3 values: `light`, `dark` and `system`, and actually sets dark or not-dark based on some separate logic)